src/SourceBuild/content/eng/finish-source-only.proj

```diff
@@ -46,6 +46,19 @@
     </PropertyGroup>
   </Target>
 
+  <!-- Write out the .version file to be bundled into the source-build tarball -->
+  <Target Name="WriteSourceBuiltSdkVersionFile"
+          AfterTargets="Build"
+          DependsOnTargets="DetermineSourceBuiltSdkNonStableVersion">
+    <!-- Content of the .version file to include in the tarball -->
+    <ItemGroup>
+      <VersionFileContent Include="$(RepositoryCommit);$(SourceBuiltSdkNonStableVersion)" />
+    </ItemGroup>
+
+    <WriteLinesToFile File="$(ArtifactsAssetsDir).version"
+                      Lines="@(VersionFileContent)" />
+  </Target>
```

Comment:
Could source-build just use the shipping .version file from the sdk folder in the SDK archive? https://github.com/dotnet/sdk/blob/4792e2a72af800ad34234f77c5248376458dabcc/src/Layout/redist/targets/GenerateLayout.targets#L38-L41

Reply:
This can for sure happen in a follow-up. I will file an issue. Note that the work to use the VMR versioning in SB is planned for P4 and being worked on as we speak.

Reply:
FWIW the version that is written into SB's .version file already comes from sdk's .version file: https://github.com/dotnet/dotnet/blob/2ba44e0dca33456408ebd3b6a6cb3ed910554078/eng/finish-source-only.proj#L40-L46

It's just the RepositoryCommit that is different. Will mention that in the issue.

Reply:
https://github.com/dotnet/source-build/issues/5068

Reply:
Yep I know. I'm just trying to get the exact same output given the inputs (to pass tests) today.